src/Service/WorkloadReportService.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace App\Service;
+
+use App\Model\Reports\WorkloadReportData;
+use App\Model\Reports\WorkloadReportWorker;
+use App\Repository\WorkerRepository;
+use App\Repository\WorklogRepository;
+
+class WorkloadReportService
+{
+    public function __construct(
+        private readonly WorkerRepository $workerRepository,
+        private readonly WorklogRepository $worklogRepository,
+        private readonly DateTimeHelper $dateTimeHelper,
+    ) {
+    }
+
+    /**
+     * Get the workload report based on the specified view mode.
+     *
+     * @param string $viewMode The view mode (default: 'week')
+     *
+     * @return WorkloadReportData The workload report data
+     *
+     * @throws \Exception When an unexpected value for viewMode is provided
+     */
+    public function getWorkloadReport(string $viewMode = 'week'): WorkloadReportData
+    {
+        // Get period based on viewmode.
+        $periods = match ($viewMode) {
```

Comment:
I think I would change all match ($viewMode) into one match and do the relevant initialisations for each of the two types.